src/compiler/types.js

```diff
@@ -27,21 +27,17 @@
  *
  * @typedef {function(!Element):*}
  */
-let BuildDomDef;
+export let BuildDomDef;
 
 /**
  * See @ampproject/bento-compiler for the definition.
  * @typedef {*}
  */
-let TreeProtoDef;
+export let TreeProtoDef;
 
 /**
- * Contains versioning data for which Runtime, if any, is present.
- * As well as the full list of (extension, version) tuples.
+ * Contains versioning data of <component, version> tuples.
  *
- * @typedef {{
- *   runtime: ?string,
- *   extensions: !Array<{name: string, version:string}>
- * }}
+ * @typedef {!Array<{name: string, version:string}>}
```

Comment:
Bikeshedding at this point, but is there a benefit to using a tuple-array here over a map/object?

Reply:
pretty equivalent. the initial structure was to mirror what appears in `head`, now that it already has more assumed pre-processing, the structure has less reasoning